ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -5363,291 +5363,52 @@ let vm_import fd printer rpc session_id params =
-              marshal fd
-                (Command (Debug ("Caught exception: " ^ Printexc.to_string e))) ;
-              marshal fd
-                (Command (PrintStderr "Failed to import directory-format XVA\n")) ;
-              debug "Import failed with exception: %s" (Printexc.to_string e) ;
-              if Client.Task.get_progress rpc session_id importtask = -1.0 then
-                Client.Task.set_status rpc session_id importtask `failure ;
-              raise (ExitWithError 2)
-            )
-          (fun () ->
-            if using_existing_task then
-              ()
-            else
-              Client.Task.destroy rpc session_id importtask
-            )
-    | Response Failed ->
-        (* possibly a Rio import *)
-        let make_command task_id =
-          let prefix = uri_of_someone rpc session_id Master in
-          let uri =
-            Printf.sprintf
-              "%s%s?session_id=%s&task_id=%s&restore=%b&force=%b&dry_run=%b%s%s"
-              prefix
-              ( if vm_metadata_only then
-                  Constants.import_metadata_uri
-              else
-                Constants.import_uri
-              )
-              (Ref.string_of session_id) (Ref.string_of task_id) full_restore
-              force dry_run
-              (if sr <> Ref.null then "&sr_id=" ^ Ref.string_of sr else "")
-              (String.concat ""
-                 (List.map (fun (a, b) -> "&vdi:" ^ a ^ "=" ^ b) vdi_map)
-              )
-          in
-          debug "requesting HttpPut('%s','%s')" filename uri ;
-          HttpPut (filename, uri)
-        in
-        let importtask =
-          if List.mem_assoc "task-uuid" params then
-            Some
-              (Client.Task.get_by_uuid rpc session_id
-                 (List.assoc "task-uuid" params)
-              )
-          else
-            None
-          (* track_http_operation will create one for us *)
-        in
-        let result =
-          track_http_operation ?use_existing_task:importtask fd rpc session_id
-            make_command "VM import"
-        in
-        let vmrefs = result |> Xmlrpc.of_string |> API.ref_VM_set_of_rpc in
-        let uuids =
-          List.map (fun vm -> Client.VM.get_uuid rpc session_id vm) vmrefs
-        in
-        let uuids =
-          if uuids = [] && dry_run then
-            ["xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"]
+    let make_command task_id =
+      let prefix = uri_of_someone rpc session_id Master in
+      let uri =
+        Printf.sprintf
+          "%s%s?session_id=%s&task_id=%s&restore=%b&force=%b&dry_run=%b%s%s"
+          prefix
+          ( if vm_metadata_only then
+              Constants.import_metadata_uri
           else
-            uuids
-        in
-        marshal fd (Command (Print (String.concat "," uuids)))
-    | _ ->
-        failwith "Thin CLI protocol error"
+            Constants.import_uri
+          )
+          (Ref.string_of session_id) (Ref.string_of task_id) full_restore
```

Comment:
This code assumes that no URL escaping is ever required. Is this true? I'm no too fond of constructing a URI this way and would have preferred constructing a map or assoc list which then some function turns into a URI, taking care of escaping.

Reply:
This is not new code. This PR mostly removes and reformats code.

Reply:
This view makes it easier to see: https://github.com/xapi-project/xen-api/pull/4539/files?diff=split&w=1